source/MaterialXRender/Util.cpp

```diff
@@ -0,0 +1,228 @@
+//
+// TM & (c) 2017 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+// All rights reserved.  See LICENSE.txt for license.
+//
+
+#include <MaterialXRender/Util.h>
+
+#include <MaterialXGenShader/Shader.h>
+#include <MaterialXGenShader/ShaderGenerator.h>
+
+namespace MaterialX
+{
+
+ShaderPtr createShader(const string& shaderName, GenContext& context, ElementPtr elem)
+{
+    if (!elem)
+    {
+        return nullptr;
+    }
+
+    context.getOptions().hwTransparency = isTransparentSurface(elem, context.getShaderGenerator());
+    return context.getShaderGenerator().generate(shaderName, elem, context);
+}
+
+ShaderPtr createConstantShader(GenContext& context,
+                            DocumentPtr stdLib,
+                            const string& shaderName,
+                            const Color3& color)
+{
+    // Construct the constant color nodegraph
+    DocumentPtr doc = createDocument();
+    doc->importLibrary(stdLib);
+    NodeGraphPtr nodeGraph = doc->addNodeGraph();
+    NodePtr constant = nodeGraph->addNode("constant");
+    constant->setParameterValue("value", color);
+    OutputPtr output = nodeGraph->addOutput();
+    output->setConnectedNode(constant);
+
+    // Generate the shader
+    ShaderPtr shader = createShader(shaderName, context, output);
+    return shader;
+}
+
+VariableBlock* getUniformBlock(ShaderPtr shader, const string& stageName, const string& blockName)
```

Comment:
I wonder if this needs to be a utility function? As far as I can see it's only used in one place inside Material. 

Also, in that particular case an exception caused by not finding the given stage or block is a fatal error (a HW shader should always have a pixel stage with public uniforms created). So I'm not sure we should hide that error by catching the exception and returning nullptr instead. The result below is that this error would never be reported since nullptr results are handled further up the chain.